File: m-solutions2020/m-solutions2020_b.cs
// https://atcoder.jp/contests/m-solutions2020/submissions/15466443

using System;
using System.Linq;

class Program
{
    static void Main()
    {
        var input = Console.ReadLine().Split().Select(int.Parse).ToArray();
        var A = input[0];
        var B = input[1];
        var C = input[2];
        var K = int.Parse(Console.ReadLine());

        var count = 0;
        while (A >= B)
        {
            count++;
            B *= 2;
        }
        while (B >= C)
        {
            count++;
            C *= 2;
        }
        Console.WriteLine(count <= K ? "Yes" : "No");
    }
}

File: m-solutions2020/m-solutions2020_b.fs
// https://atcoder.jp/contests/m-solutions2020/submissions/15462454

let [| A; B; C |] = stdin.ReadLine().Split() |> Array.map int
let K = stdin.ReadLine() |> int

let rec solver x y i =
    if x < y then i
    else solver x (y * 2) (i + 1)

let s1 = solver A B 0
let s2 = solver (B <<< s1) C 0

if s1 + s2 <= K then "Yes"
else "No"
|> stdout.WriteLine
